components/base/kinematicbase/ptgKinematics.go

```diff
@@ -177,10 +177,11 @@ func (ptgk *ptgBaseKinematics) ExecutionState(ctx context.Context) (motionplan.E
 		return motionplan.ExecutionState{}, errors.New("cannot call ExecutionState on a base without a localizer")
 	}
 	ptgk.inputLock.RLock()
+	defer ptgk.inputLock.RUnlock()
```

Comment:
We don't want this change; we want to release the RLock as soon as possible, rather than waiting on CurrentPosition to return.

Instead; move the `actualPIF, err := ptgk.CurrentPosition(ctx)` above the RLock section, and then move the `currentPlan := ptgk.stepsToPlan(currentExecutingSteps, actualPIF.Parent())` into the RLock block